SalesforceSDK/Core/Auth/AccountManager.cs

```diff
@@ -37,13 +37,33 @@
 namespace Salesforce.SDK.Auth
 {
     /// <summary>
-    ///     Class providing (static) methods for creating/deleting or retrieving an Account
+    /// Class providing (static) methods for creating/deleting or retrieving an Account
     /// </summary>
     public class AccountManager
     {
         private static IAuthHelper AuthStorageHelper => SDKServiceLocator.Get<IAuthHelper>();
         private static ILoggingService LoggingService => SDKServiceLocator.Get<ILoggingService>();
 
+        private static Account _loggedInAccount;
+        private static Account LoggedInAccount
+        {
+            set
+            {
+                var oldAccount = _loggedInAccount;
+                _loggedInAccount = value;
+
+                // Raise the event in AccountManager if this is a different account.
+                // This check is necessary as sometimes CurrentAccount.Set is called
+                // even if the same account was already set, so use the unique combo of
+                // InstanceUrl and UserId to tell if the account has actually changed (login/logout).
+                if (oldAccount?.InstanceUrl != value?.InstanceUrl && oldAccount?.UserId != value?.UserId)
```

Comment:
I'd prefer the standard approach of comparing ```orgId``` and ```userId```, as opposed to ```instanceUrl```. ```instanceUrl``` is transient (it can change between refresh sessions during org migration).